docs/testing-guidelines/TestRoadmap.md

```diff
@@ -0,0 +1,193 @@
+Over time, the number of `Pending` tests should drive to zero, and we should be tracking the list of tests which are marked `pending` and track those to be sure that they do not spend too long in this state.
+`Skipped` tests due to platform applicability are certainly valid, and it is important that we track how many tests are skipped and for what reason.
+A test which is skipped for all platforms should be considered carefully as to whether it should exist at all.
+In either case, `pending` or `skipped` tests should be tracked over time so we can determine whether we are improving our ability to measure quality.
+
+In the best case, the _total_ number of tests is the same across all platforms.
+The count of Skipped/Pending tests would naturally be different as not all tests will be applicable to every platform, but if we can have consistency of test count across platforms, it will be easier to compare test results on one platform vs another.
+
+## Remoting Considerations
+Given that one of the targeted scenarios is to manage heterogeneous environments remotely, we need a test environment which encompasses the available platforms and protocols.
+Our current test infrastructure does not have comprehensive support for remote testing except for loopback operations.
+We need a matrix of protocols and platforms to ensure that we have adequate coverage to ensure quality.
+
+In addition to loopback tests using both WSMan and SSH protocols, we should have the following _minimum_ matrix of client/server connections for both WSMan and SSH (where available) protocols.
+* Windows Client->Nano Server
+* Windows Client->Linux Server
+* Linux Client -> Windows Server
+* OSX Client -> Nano Client
+* PowerShell Core Client -> Full PowerShell Server
+* Full PowerShell Client -> PowerShell Core Server
+* Downlevel Full PowerShell Client -> PowerShell Core Server 
+
+### Challenges
+* On Windows, we have cmdlets to enable and disable remote access, those cmdlets do not exist on non-Windows platforms (they rely on configuration stored in the registry), nor do we support configuration for the SSH protocol.
+We need to be sure that we can easily enable remoting for the non-Windows platforms and support both WSMan _and_ SSH configurations.
+* Our current multi-machine tests do not test the connection code, they simply execute test code remotely and retrieve results and assume a good connection.
+The infrastructure used for these tests is STEX which is not an open environment.
+We will need to create automation to create and configure the test systems in the test matrix and then invoke tests on them.
+It is not clear that our current CI systems can accomodate our needs here as neither AppVeyor or Travis can supply us with all of the OS images needed.
+We may need to create our own heterogeneous environment in Azure, or look to other teams (MS Build Lab) for assistance.
```

Comment:
We should talk to Bryce and find out about the Jenkins system, it may be able to support our needs.